packages/eslint-plugin-vtex/tests/lib/rules/prefer-use-effect-named-callback.test.js

```diff
@@ -0,0 +1,103 @@
+const { RuleTester } = require('eslint')
+
+const rule = require('../../../lib/rules/prefer-use-effect-named-callback')
+
+const ruleTester = new RuleTester()
+
+const message = 'Prefer useEffect with named function or constant callbacks.'
+const arrowFunctionError = {
+  message,
+  type: 'ArrowFunctionExpression',
+}
+
+const functionExpressionError = {
+  message,
+  type: 'FunctionExpression',
+}
+
+ruleTester.run('prefer-use-effect-named-callback', rule, {
+  valid: [
+    {
+      code: `function Component() {
+        useEffect(function namedEffect() {}, [])
```

Comment:
Maybe add a test for this case:

```ts
useEffect(debounce(() => {}, 200), [])
```

Today the implementation will accept this kind of input and it makes sense because you can't identify which argument gonna be enhanced on higher-order functions, but it is nice to have the test case to identify that this is valid today.

Reply:
Ok ok, i see. I will add it, thank you